contracts/Vault.sol

```diff
@@ -694,6 +713,27 @@ contract Vault is
     // Internal API
     //
 
+    function _immediateInvestment() private {
+        (uint256 maxInvestableAmount, uint256 alreadyInvested) = investState();
+
+        if (maxInvestableAmount <= alreadyInvested) return;
```

Comment:
this is not fully covered in tests, consider adding a test case for `maxInvestableAmount <= alreadyInvested == true`